### PR TITLE
Expand local development section of README.md

### PR DESCRIPTION
**Pull request by lucaswerkmeister**
Split it into the one-time setup and the day-to-day work, and provide some more details on both.

Bug: T302321

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,9 +8,28 @@ Please report bugs on [phabricator](https://phabricator.wikimedia.org/project/vi
...
+e.g. `npm run test-only` to run only the tests without the linters,
+or `npm run test:unit` to run only the unit tests.
+Use `npm run fix` to have the linters fix any fixable errors.
+
+Run `npm run dev` to run the dev entry point, then visit it at <http://localhost:3000/>.
+Changes to the source files will be applied automatically,
+so you can start this once and then leave it running in the background.
+
+For instructions to see the form in the context of the special page,
+see the [WikibaseLexeme readme](https://gerrit.wikimedia.org/g/mediawiki/extensions/WikibaseLexeme/+/master/README.md).
```
[the remark this answers is not in the document]
Comment (itamargiv) [reply README.md:32]: Looking great, thank you! I think maybe making it less prosaic and more reference oriented can help with discoverability. What I mean by that is give each point or script in the workflow its own heading and use full-width code boxes to make the commands more prominent. Some examples:

- Mismatch Finder (Our own ;) ): https://github.com/wmde/wikidata-mismatch-finder/blob/main/docs/README.md#testing
- Rust Docs (one of my favs): https://doc.rust-lang.org/book/ch01-01-installation.html
- Any Community tutorial by digital ocean: https://www.digitalocean.com/community/tutorials/how-to-build-and-install-go-programs

File: README.md [hunk 2]
```diff
@@ -8,9 +8,42 @@ Please report bugs on [phabricator](https://phabricator.wikimedia.org/project/vi
 
 _**Prerequisites: This repository requires node v16 and up**_
 
+### One-time setup
```
Comment (micgro42) [inline README.md:11]: "One-time setup" sounds a bit strange to me. How about "Initial setup"?

File: README.md [hunk 3]
```diff
@@ -8,9 +8,28 @@ Please report bugs on [phabricator](https://phabricator.wikimedia.org/project/vi
...
+Use `npm run fix` to have the linters fix any fixable errors.
+
+Run `npm run dev` to run the dev entry point, then visit it at <http://localhost:3000/>.
+Changes to the source files will be applied automatically,
+so you can start this once and then leave it running in the background.
+
+For instructions to see the form in the context of the special page,
+see the [WikibaseLexeme readme](https://gerrit.wikimedia.org/g/mediawiki/extensions/WikibaseLexeme/+/master/README.md).
 
 ## Deploy Local Commits
```
[the remark this answers is not in the document]
Comment (micgro42) [reply README.md:48]: yeah, let not touch that section here

**Review by itamargiv — DISMISSED** [review]
Thanks, this is much more spaced out and prominent, I'd still like to add more headings so that we can link to specific sections or create a TOC for people to skim over. But I agree that this can be deferred to later, when we grow this README a little further.

File: README.md (hunk 2 above)
Comment (lucaswerkmeister) [reply README.md]: Sure, done.

**Pull request merged**